sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructField.scala

```diff
@@ -49,7 +49,7 @@ case class StructField(
   }
 
   // override the default toString to be compatible with legacy parquet files.
-  override def toString: String = s"StructField($name,$dataType,$nullable)"
+  override def toString: String = s"""StructField("$name",$dataType,$nullable)"""
```

Comment:
Scala sides doesn't have `repr` contract to make it re-construct-able like Python sides.
Also, I think this can't handle `"` character in the middle of its name, for instance.

Reply:
I tested the code to confirm this works.  The """ allows for the embedded " to work correctly.

True, Scala doesn't have a repr contract for runnable code.  However, having toString produce runnable code here has a real use-case for users.  Users can inferSchema, get the generated schema code, tweak it as needed, and then provide the schema in the future.  I've needed this many times in my projects.  I'll make sure I add this to the PR comment.  And also open a JIRA.